gulpfile.js/build.js

```diff
@@ -0,0 +1,411 @@
+
+// The Google Cloud Storage bucket used to store build job artifacts
+const TRAVIS_GCS_PATH = 'gs://amp-dev-ci/travis/';
+// Branches that should actually trigger a build
+const BUILDING_BRANCHES = ['future'];
+
+
+/**
+ * Cleans all directories/files that get created by any of the following
+ * tasks
+ *
+ * @return {Promise}
+ */
+function clean() {
+  return del([
+    project.absolute('.cache/**/*'),
+
+    project.paths.DIST,
+    project.paths.BUILD,
+
+    project.absolute('boilerplate/dist'),
+
+    project.paths.CSS,
+    project.absolute('pages/extensions/**/*.pyc'),
+    project.absolute('pages/content/amp-dev/documentation/examples/documentation/**/*.html'),
+    project.absolute('pages/content/amp-dev/documentation/examples/previews/**/*.html'),
+    project.absolute('pages/icons'),
+    project.absolute('pages/layouts'),
+    project.absolute('pages/macros'),
+    project.absolute('pages/views'),
+    project.absolute('pages/.depcache.json'),
+    project.absolute('pages/podspec.yaml'),
+
+    project.paths.GROW_BUILD_DEST,
+    project.paths.STATICS_DEST,
+    project.absolute('platform/static'),
+
+    project.absolute('playground/dist'),
+  ], {'force': true});
+}
+
+
+/**
+ * Compiles all SCSS partials to CSS
+ *
+ * @return {Stream}
+ */
+function sass() {
+  const options = {
+    'outputStyle': 'compressed',
+    'includePaths': project.paths.SCSS,
+  };
+
+  return gulp.src(`${project.paths.SCSS}/**/[^_]*.scss`)
+      .pipe(gulpSass(options))
+      .on('error', function(e) {
+        console.error(e);
+        // eslint-disable-next-line no-invalid-this
+        this.emit('end');
+      })
+      .pipe(gulp.dest(project.paths.CSS));
+}
+
+/**
+ * Copies the templates into the Grow pod
+ *
+ * @return {Stream}
+ */
+function templates() {
+  return gulp.src(`${project.paths.TEMPLATES}/**/*`)
+      .pipe(gulp.dest(project.paths.GROW_POD));
+}
+
+/**
+ * Copies the icons into the Grow pod
+ *
+ * @return {Stream}
+ */
+function icons() {
+  return gulp.src(`${project.paths.ICONS}/**/*`)
+      .pipe(gulp.dest(`${project.paths.GROW_POD}/icons`));
+}
+
+/**
+ * Runs all tasks needed to build the frontend
+ * @return {undefined}
+ */
+function buildFrontend(done) {
+  return (gulp.parallel(sass, templates, icons))(done);
+}
+
+/**
+ * Builds the playground
+ * @return {Promise}
+ */
+async function buildPlayground() {
+  await sh('mkdir -p playground/dist');
+  return sh('npm run build:playground');
+}
+
+/**
+ * Builds the boilerplate generator
+ * @return {Promise}
+ */
+function buildBoilerplate() {
+  return sh('node build.js', {
+    workingDir: project.absolute('boilerplate'),
+  });
+}
+
+
+/**
+ * Builds documentation pages, preview pages and source files by parsing
+ * the samples sources
+ *
+ * @return {Promise}
+ */
+function buildSamples() {
+  return samplesBuilder.build(true);
+}
+
+
+/**
+ * Runs all importers
+ *
+ * @return {Promise}
+ */
+function importAll() {
+  return Promise.all([
```

Comment:
Wanted to migrate this to `gulp.parallel` as you suggested but it seems to fiddle with the scope of the functions which made them fail.